addon/webextension/background/auth.js

```diff
@@ -0,0 +1,208 @@
+/* globals browser */
+/* globals main, makeUuid, deviceInfo, analytics, catcher, defaultSentryDsn, communication */
+
+window.auth = (function () {
+  let exports = {};
+
+  let registrationInfo;
+  let initialized = false;
+  let authHeader = null;
+  let sentryPublicDSN = null;
+  let abTests = {};
+
+  catcher.watchPromise(browser.storage.local.get(["registrationInfo", "abTests"]).then((result) => {
+    if (result.abTests) {
+      abTests = result.abTests;
+    }
+    if (result.registrationInfo) {
+      registrationInfo = result.registrationInfo;
+    } else {
+      registrationInfo = generateRegistrationInfo();
+      console.info("Generating new device authentication ID", registrationInfo);
+      return browser.storage.local.set({registrationInfo});
```

Comment:
This particular promise chain only runs when auth.js is initially loaded. I don't think we would ever encounter a race condition here.